kava-stats.js

```diff
@@ -335,92 +336,52 @@ const mapDenomTotalSupplied = async (denoms, siteData) => {
 
     coins[denom] = denomTotalSupplyValue
   }
-
+ 
   return coins
 }
 
-const mapPlatformAmounts = async (denoms, platformAmounts) => {
-  const coins = {};
-
-  for (const denom of denoms) {
-    const coin = { locked: 0, borrowed: 0, fees: 0 };
-    const platformAmount = platformAmounts[denom];
-
-    if(platformAmount) {
-      const { collateral, principal, accumulated_fees } = platformAmount.reduce(function(accumulator, item) {
-        Object.keys(item.cdp).forEach(function(key) {
-          let c = ['collateral', 'principal', 'accumulated_fees'];
-          if(c.includes(key)){
-            accumulator[key] = Number((accumulator[key] || 0)) + Number(item.cdp[key].amount);
-          }
-        });
-        return accumulator;
-      }, {});
-
-      coin['locked'] = isKavaNativeAsset(denom) ? Number(collateral/FACTOR_SIX) : Number(collateral/FACTOR_EIGHT)
-      coin['borrowed'] = Number(principal/FACTOR_SIX)
-      coin['fees'] = Number(accumulated_fees/FACTOR_SIX)
-    }
-    coins[denom] = coin;
-  }
-  return coins;
+const mapPlatformAmounts = async (totalCollateral, totalPrincipal) => {
+    const coins = {};
+      for (const denom of totalCollateral) {
+        const { amount: { amount }, collateral_type } = denom;
+        coins[collateral_type] = {};
+        coins[collateral_type].collateral = amount;
+      };
+      for (const denom of totalPrincipal) {
+        const {collateral_type, amount: { amount }} = denom;
+        coins[collateral_type].principal = amount; 
+      };
+     return coins;
 };
 
 const mapCdpParams = async (denoms, cdpParamsData) => {
   const coins = {};
-
   const mappedLimits = {};
   const mappedStabilityFees = {};
   let usdxDebtLimit = 0;
   if(cdpParamsData) {
     for (const denom of cdpParamsData.collateral_params) {
-
       const debtLimit = denom.debt_limit ? Number(denom.debt_limit.amount)/FACTOR_SIX : 0;
       mappedLimits[denom.type] = { debtLimit }
-
       const secondsPerYear = 31536000;
       const stabilityFeePercentage = ((Number(denom.stability_fee) ** secondsPerYear - 1) * 100).toFixed(2);
       mappedStabilityFees[denom.type] = stabilityFeePercentage
-    }
-
+    };
     usdxDebtLimit = Number(cdpParamsData.global_debt_limit.amount)/FACTOR_SIX;
-  }
-
+  };
   for (const denom of denoms) {
     let limit = 0;
-    let stabilityFee = ' ';
-
     if (denom === 'usdx') {
       limit = usdxDebtLimit
     } else {
       let cdpParam = mappedLimits[denom]
       if(cdpParam) { limit = cdpParam.debtLimit }
-    }
-
-    coins[denom] = { debtLimit: limit, stabilityFeePercentage: mappedStabilityFees[denom] }
-  }
+    };
+    coins[denom] = { debtLimit: limit, stabilityFeePercentage: mappedStabilityFees[denom] || '0.00'}
```

Comment:
@downing034 - interested in your take on if this is just as good an empty record/guard as something like

```
let stabilityFeePercentage = '0';
if (mappedStabilityFees[denom]) {
  stabilityFeePercentage = mappedStabilityFees[denom];
}
coins[denom] = { debLimit: limit, stabilityFeePercentage }